### PR TITLE
chore: update cargo deny to handle deprecated config entries

### PR DESCRIPTION
**Pull request by TomAFrench**
A bunch of the [cargo deny config has been deprecated](https://github.com/EmbarkStudios/cargo-deny/pull/606) so this PR updates to target the new spec.

I don't know if you want to blanket accept MPL2.0 so I've just added exceptions for the current dependencies.

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -65,6 +59,12 @@ exceptions = [
     { allow = ["CC0-1.0"], name = "tiny-keccak" },
     { allow = ["CC0-1.0"], name = "more-asserts" },
     { allow = ["CC0-1.0"], name = "aurora-engine-modexp" },
+    # TODO: decide on MPL-2.0 handling
+    # These dependencies are grandfathered in in https://github.com/paradigmxyz/reth/pull/#
```
Comment (TomAFrench) [inline deny.toml:63]:
```suggestion
    # These dependencies are grandfathered in in https://github.com/paradigmxyz/reth/pull/6980
```

File: deny.toml [hunk 2]
```diff
@@ -48,8 +44,6 @@ allow = [
     "Unicode-3.0",
     # https://github.com/briansmith/ring/issues/902
     "LicenseRef-ring",
-    # https://github.com/briansmith/webpki/issues/148
-    "LicenseRef-webpki",
```
Comment (DaniPopes) [inline deny.toml:52]: Please also remove the clarify below

File: deny.toml [hunk 3]
```diff
@@ -65,6 +59,12 @@ exceptions = [
     { allow = ["CC0-1.0"], name = "tiny-keccak" },
     { allow = ["CC0-1.0"], name = "more-asserts" },
     { allow = ["CC0-1.0"], name = "aurora-engine-modexp" },
+    # TODO: decide on MPL-2.0 handling
+    # These dependencies are grandfathered in in https://github.com/paradigmxyz/reth/pull/6980
+    { allow = ["MPL-2.0"], name = "attohttpc" },
+    { allow = ["MPL-2.0"], name = "option-ext" },
+    { allow = ["MPL-2.0"], name = "webpki-roots" },
+
```
Comment (DaniPopes) [inline deny.toml:67]:
```suggestion
```

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: deny.toml (hunk 2 above)
Comment (TomAFrench) [reply deny.toml:52]: Thanks, done now.

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Pull request merged**